docs: improve description of committing in contributing (#404)
cognitedata/cognitesdk-js: CONTRIBUTING.md

The commit messages are used to automatically bump the versions of changed packages, and write automatic changelogs.

We use semantic versioning, with versions `MAJOR.MINOR.PATCH`.

 - For fixes, start the commit with `fix: `. This will bump PATCH.
 - For features, start the commit with `feat: `. This will bump MINOR.
 - For changes that break backwards compatibility, add a `!` before the colon: `feat!: `.
   This will bump MAJOR version.
   
   The same can be achieved by specifying a footer.
   The footer must come after a blank line, and start with `BREAKING CHANGE: ` (colon mandatory),
   followed by a description of what breaks.
 - For extra details in the changelog, you can specify a scope like so: `feat(assets): `.
 - For other changes there are types without version bumping semantics:
   - `docs: ` changes to documentation
   - `build: ` changes to build scripts and config
   - `ci: ` changes to ci scripts and pipeline
   - `refactor: ` code moving and renaming
   - `style: ` fixes to code style
   - `perf: ` changes to improve performance
   - `revert: ` changing things back 
   - `chore: ` miscelanious changes

#### Example
```

> docs(changelog): add example of commit with subject line

Reviewer: Would you ever have "changelog" here?

> docs: add example of commit with subject line to CHANGELOG

Reviewer 2: ```suggestion
docs(changelog): add example of commit with subject line
```

We use semantic versioning, with versions `MAJOR.MINOR.PATCH`.

 - For fixes, start the commit with `fix: `. This will bump PATCH.
 - For features, start the commit with `feat: `. This will bump MINOR.

>  - For changes that break backwards compatability, add a `!` before the colon: `feat!: `.

Reviewer 2: ```suggestion
 - For changes that break backwards compatibility, add a `!` before the colon: `feat!: `.
```

   
   Alternativly you may specify a footer (or both).
   The footer must come after a blank line, and start with `BREAKING CHANGE:` (colon mandatory),
   followed by a description of what breaks.

>  - For extra details in the changelog, you can specify a scope, e.g. `feat(assets):`.

Reviewer 2: ```
For other changes, use `refactor:`, `test:`, `docs:`, `chore:` etc.
Try something out, and the commit hook will tell you what's allowed.
```
maybe we can explain more here?
like chore is for changes related to project setup, etc

and provide a full list of options

#### Example
```

> docs(changelog): add example of commit with subject line

Reviewer 2: ```suggestion
docs(contributing-readme): add example of commit with subject line
```
